bin/cron/pardot_import

```diff
@@ -0,0 +1,212 @@
+  prospects_payload_json_encoded = prospects_payload_json_encoded.gsub("+", "%2B")
+
+  "#{PARDOT_BATCH_CREATE_URL}?prospects=#{prospects_payload_json_encoded}"
+end
+
+def post_with_auth_retry(url)
+  # do the post to Pardot
+  post_request(url)
+rescue InvalidApiKeyException
+  # If we fail with an invalid key, that probably means our API key has expired. Try one time
+  # to request a new API key and try the post again. If that fails, that is a fatal error.
+  request_pardot_api_key
+  post_request(url)
+end
+
+# Make an API request
+def post_request(url)
+  request_pardot_api_key if $api_key.nil?
+  # add the API key and user key parameters to the URL
+  auth_url = append_auth_params_to_url(url)
+  uri = URI(auth_url)
+  response = Net::HTTP.post_form(uri, {})
+
+  puts "Response: " + response.to_s
+
+  xml = Nokogiri::XML(response.body, &:noblanks)
+  raise "Pardot create user response did not return parsable XML" if xml.nil?
+
+  error_details = xml.xpath('/rsp/err').text
+  raise InvalidApiKeyException if error_details.include? "Invalid API key or user key"
+
+  status = xml.xpath('/rsp/@stat').text
+  raise "Pardot create user response did not include status" if status.nil?
+
+  raise "Pardot create user response failed with status " + status + " " + xml.to_s if status != "ok"
+end
+
+def append_auth_params_to_url(url)
+  "#{url}&api_key=#{$api_key}&user_key=#{CDO.pardot_user_key}"
+end
+
+def main
+  return unless only_one_running?(__FILE__)
+  prospects = []
+
+  # query the conact rolliups
+  PEGASUS_DB[:contact_rollups].where("roles like '%Teacher%' AND pardot_sync_at IS NULL").each do |contact_rollup|
+    puts contact_rollup[:email]
+
+    # Pardot explicitly considers invalid any email addresses that end with a period in the base
+    # (before the @). We have at least one of these, which appears to be a typo.
+    next if contact_rollup[:email].include? ".@"
+
+    # Map database field names and data to Pardot fields
+    prospect = {}
+    MYSQL_TO_PARDOT_MAP.each do |mysql_key, pardot_info|
+      db_value = contact_rollup[mysql_key]
+      if db_value.presence
+        if pardot_info[:multi]
+          # For multi data fields (multiselect,etc ), we set value names as [fieldname]_0, [fieldname]_1, etc
+          values = db_value.split(",")
+          i = 0
+          values.each do |value|
+            prospect["#{pardot_info[:field]}_#{i}"] = value
+            i += 1
+          end
+        else
+          # for single data fields, just set [fieldname] = value
+          prospect[pardot_info[:field]] = db_value
+        end
+      end
+    end
+    # special case: if contact has opted out, set the two different Pardot flavors of opted out
+    # to true. Also, only ever set this to true, otherwise set no value; never set it back
+    # to false. Pardot is the authority on opt-out data, so never reset any opt-out setting it has stored.
+    if contact_rollup[:opted_out] == true
```

Comment:
Can the comparison with `true` be removed? If not, an explanation as to why truthy values should be considered false is warranted.